### DISCUSSION TITLE
Apple M1 Results

### DISCUSSION THREAD

**Discussion by DennisFaucher** [rasbt/machine-learning-notes#26] [Show and tell]
I just created a new conda 3.9 environment, activated it, used conda to install numpy, pytorch, torchvision. Here are my results on a Mac Mini M1:

python lenet-mnist.py --device "mps"

Epoch: 001/001 | Batch 0000/0421 | Loss: 2.3098
Epoch: 001/001 | Batch 0100/0421 | Loss: 0.2646
Epoch: 001/001 | Batch 0200/0421 | Loss: 0.1437
Epoch: 001/001 | Batch 0300/0421 | Loss: 0.1010
Epoch: 001/001 | Batch 0400/0421 | Loss: 0.0734
Time / epoch without evaluation: 0.15 min
Epoch: 001/001 | Train: 97.32% | Validation: 97.77% | Best Validation (Ep. 001): 97.77%
Time elapsed: 0.19 min
Total Training Time: 0.19 min
Test accuracy 97.40%
Total Time: 0.20 min

(GPU was definitely used)
![Screen Shot 2022-09-20 at 10 40 54 AM](https://user-images.githubusercontent.com/9034190/191288292-24fadeb9-678e-4abe-97de-4a26b5687916.png)

**Comment by DennisFaucher** [comment]
❯❯python mlp-mnist.py --device "mps"
torch 1.12.1
device mps
Epoch: 001/001 | Batch 0000/0421 | Loss: 2.3063
Epoch: 001/001 | Batch 0100/0421 | Loss: 0.3429
Epoch: 001/001 | Batch 0200/0421 | Loss: 0.3100
Epoch: 001/001 | Batch 0300/0421 | Loss: 0.3713
Epoch: 001/001 | Batch 0400/0421 | Loss: 0.3542
Time / epoch without evaluation: 0.07 min
Epoch: 001/001 | Train: 91.67% | Validation: 93.42% | Best Validation (Ep. 001): 93.42%
Time elapsed: 0.11 min
Total Training Time: 0.11 min
Test accuracy 92.20%
Total Time: 0.12 min

**Comment by DennisFaucher** [comment]
python vgg16-cifar10.py --device "mps"
torch 1.12.1
device mps
Epoch: 001/001 | Batch 1400/1406 | Loss: 2.2369
Time / epoch without evaluation: 104.24 min
Epoch: 001/001 | Train: 15.99% | Validation: 15.42% | Best Validation (Ep. 001): 15.42%
Time elapsed: 122.93 min
Total Training Time: 122.93 min
Test accuracy 16.19%
Total Time: 126.78 min

Experienced some "Error: command buffer exited with error status. The Metal Performance Shaders operations encoded on it may not have completed." Might be the conda version of pytorch.
